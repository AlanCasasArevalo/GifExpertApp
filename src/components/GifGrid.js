import React from 'react'
import PropTypes from 'prop-types'

const GifGrid = ({category}) => {
    // const [inputValue, setInputValue] = useState('')

    const getGifs = async () => {
        const apiKey = 'irpbY8APMlGjv0zXNFjdWqDFsvoCiruB';
        const urlToCall = `http://api.giphy.com/v1/gifs/search?q=${category}&limit=10&api_key=${apiKey}`
        const resp = await fetch(urlToCall);
        const { data } = await resp.json();

        const gifs = data.map( image => {
            return {
                id: image.id,
                title: image.title,
                url: image.images?.downsized_medium.url
            }
        })

        console.log(``, data)

    }

    getGifs()

    const handleInputValueChange = (e) => {
        // setInputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        // e.preventDefault()
        // if (inputValue.trim().length > 2) {
        //     setCategories(categories => [...categories, inputValue])
        //     setInputValue('')
        // }
    }

    return (
        <>
            <h3>{category}</h3>
        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid;
