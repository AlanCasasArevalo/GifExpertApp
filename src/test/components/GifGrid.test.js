import {shallow} from "enzyme";
import React from "react";
import GifGrid from "../../components/GifGrid";

describe('GifGrid Component test', () => {

    const category = 'Hola'
    let wrapper = shallow(
        <GifGrid
            category={category}
        />)


    beforeEach(() => {
        wrapper = shallow(
            <GifGrid
                category={category}
            />)
    })

    test('GifGrid should be defined', () => {
        expect(<GifGrid
            category={category}
        />).toBeDefined()
    })
    test('GifGrid should be an object', () => {
        expect(typeof <GifGrid
            category={category}
        />).toBe('object')
    })

    test('GifGrid should return a snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    describe('H3 CATEGORY TAG', () => {
        test('It should get animate__flash class', () => {
            const h3 = wrapper.find('h3')
            const {className, children} = h3.props()
            expect(className).toStrictEqual('animate__animated animate__flash')
        })

        test('It should get category value', () => {
            const h3 = wrapper.find('h3')
            const {className, children} = h3.props()
            expect(children).toStrictEqual(category)
        })
    })

    test('p loading text should be Cargando ....', () => {
        const p = wrapper.find('p')
        const {className, children} = p.props()
        expect(children).toStrictEqual('Cargando ....')
    })

    describe('DIV CATEGORY TAG', () => {
        test('div should get card-grid class', () => {
            const div = wrapper.find('div')
            const {className, children} = div.props()
            expect(className).toStrictEqual('card-grid')
        })
    })
})
