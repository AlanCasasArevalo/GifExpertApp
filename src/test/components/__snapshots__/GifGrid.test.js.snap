// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid Component test GifGrid should return a snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flash"
  >
    Hola
  </h3>
  <p>
    'Cargando ....'
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
